### DISCUSSION TITLE
Input on 6.3

### DISCUSSION THREAD

**Discussion by QORTEC** [Owl-Maintain/brlaser#10] [Ideas]
I'm looking for feedback on some projects & changes I've been considering:

---

### Test Driver?
I'm thinking of adding a "brlaser _test_ driver", it's currently listed under:
Make > `Brother`
Model > `brlaser test driver` | `brlaser duplex test driver`

The rationale behind this addition is to provide users with a clearly labeled test driver option, allowing them to confidently check for compatibility. Additionally, this driver enables `300dpi` resolution, encouraging users to test printing at that setting as well.

> If your printer is not officially supported, you can try selecting any driver marked as brlaser to test if the driver will work for your printer.
`README.md`

This addition with a revelant update to the `README` should reduce the instances of end users select a simplex driver (for initial testing) and then report issues or complaints about the lack of duplex or `300dpi` printing support.

---

### Validation Script?
Historically, "errors" within new printer definitions have slipped into the driver unnoticed. To address this, I have created a bash script that examines `brlaser.drv.in` for errors and proposes corrections via a `diff`.
The script currently:
 - Ensures that driver entries follow to the standard conventions
 - Lists driver entries in alphanumeric order
 - Checks for duplicate `PCFileName` entries

Additionally, I'm exploring the possibility of adding an additional a check to identify unsupported printer types (such as color, inkjet, label printers).

---

### Probe Script?
Should we create/place a `brprobe` bash script in `/usr/bin`?

Rather than requiring end users to run `lpinfo`, it might be more intuitive create "_probe_" command.
This script would executing `lpinfo -l -v`, then search for printers using the `usb` or `lpd` protocols, finally output relevant information from `lpinfo` alongside a properly formatted printer definition.

The advantage of the script is that
- End users can get the necessary information to enable driver support without needing to connect their printer via USB.
- The script trims down the (verbose) output of `lpinfo -l -v` providing only the necessary information.
- Advanced users have a properly formatted printer definition for their pull request

---

### Rename The Driver?
Should we change the driver name from `brlaser` to `Owl-Maintain/brlaser`? Would this change be warranted and appropriate?

For Example:
- From: `Brother HL-2270DW series, using brlaser v6`
- To:   `Brother HL-2270DW series, using Owl-Maintain/brlaser v6`

In terms of functionality, the **Owl-Maintain** fork of **brlaser** only has a minor code modification, that specifically addresses duplex printing. Consequently, any driver-related issues within our fork are likely to affect the upstream `pdewacht:master`.
The main difference from upstream lies in our expanded list of printer definitions, which doesn't alter the core functionality of the driver rather, it enhances the user experience by simplifying printer setup by adding CUPS printer entries.

The **Owl-Maintain** fork is not expected to undergo significant divergence in terms of code (core functionality), unless a new contributor emerges to actively maintain this aspect of the fork. So, would it be appropriate to modify the name when there hasn't been (nor is there expected to be) a substantial functional code change?

Conversely, upstream development appears to have slowed (perhaps even stalled), with issues and pull requests remaining unanswered. **Owl-Maintain** consistently monitors upstream Git issues and pull requests to assess whether any action should be taken. Additionally, users have been referring to **brlaser** `v6.x` in their discussions on upstream issues, despite the last official release being `v6`. This suggests that users are actually using the **Owl-Maintain** fork and are still submitting issues upstream.

The question basically boils down to; whether whether the extent of our modifications justifies a renaming or rebranding, and whether redirecting traffic (in terms of issues and bug reports) to our fork would have a negative impact on the upstream project, especially if it becomes active again in the future.

---

Any insights would be appreciated!

`Owl-Maintain/brlaser`; @QORTEC

**Comment by qwertychouskie** [comment]
All the changes sound reasonable, though honestly I'd suggest calling the release `v7` rather than `v6.3`.

**Reply by QORTEC** [reply]
In the beginning I started using `v6.x` since:
 - my fork (originally) _only_ added new printer definitions
 - upstream never used point releases, this would be unique to the fork
 - i originally was thinking of `v7` as the next official upstream release

I'll definitely think on whether I'll use `v7`; There are definitely a few more changes/additions that I would like to do before bumping a whole version.
